homeassistant/components/aurora_abb_powerone/sensor.py

```diff
@@ -61,70 +70,43 @@ async def async_setup_entry(
     """Set up aurora_abb_powerone sensor based on a config entry."""
     entities = []
 
-    client = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
     data = config_entry.data
 
     for sens in SENSOR_TYPES:
-        entities.append(AuroraSensor(client, data, sens))
+        entities.append(AuroraSensor(coordinator, data, sens))
 
     _LOGGER.debug("async_setup_entry adding %d entities", len(entities))
     async_add_entities(entities, True)
 
 
-class AuroraSensor(AuroraEntity, SensorEntity):
-    """Representation of a Sensor on a Aurora ABB PowerOne Solar inverter."""
+class AuroraSensor(CoordinatorEntity[AuroraAbbDataUpdateCoordinator], SensorEntity):
+    """Representation of a Sensor on an Aurora ABB PowerOne Solar inverter."""
 
     _attr_has_entity_name = True
 
     def __init__(
         self,
-        client: AuroraSerialClient,
+        coordinator: AuroraAbbDataUpdateCoordinator,
         data: Mapping[str, Any],
         entity_description: SensorEntityDescription,
     ) -> None:
         """Initialize the sensor."""
-        super().__init__(client, data)
+        super().__init__(coordinator)
+        self._data = data
```

Comment:
You're not using this anywhere else now so this can be removed

Reply:
Yes, agreed, done.